src/providers/dart_highlighting_provider.ts

```diff
@@ -0,0 +1,69 @@
+"use strict";
+
+import {
+	DocumentHighlightProvider, DocumentHighlight, TextDocument, Position, CancellationToken, Range
+} from "vscode";
+import { Analyzer } from "../analysis/analyzer";
+import * as as from "../analysis/analysis_server_types";
+
+export class DartDocumentHighlightProvider implements DocumentHighlightProvider {
+	private analyzer: Analyzer;
+	constructor(analyzer: Analyzer) {
+		this.analyzer = analyzer;
+	}
+
+	provideDocumentHighlights(
+		document: TextDocument, position: Position, token: CancellationToken
+	): Thenable<DocumentHighlight[]> {
+		let file = document.fileName;
+		let offset = document.offsetAt(position);
+
+		this.analyzer.analysisSetSubscriptions({ subscriptions: { "OCCURRENCES": [file] } });
+
+		return new Promise<DocumentHighlight[]>((resolve, reject) => {
+			let disposable = this.analyzer.registerForAnalysisOccurrences(n => {
+				if (n.file != file)
+					return;
+
+				this.analyzer.analysisSetSubscriptions({ subscriptions: { "OCCURRENCES": [] } });
+				disposable.dispose();
+
+				let highlights: DocumentHighlight[] = [];
+				for (let occurrence of n.occurrences) {
```

Comment:
I'm not familiar with this - is it expected to take just the first occurrence and not all of them?